File: mall-admin-backend/src/main/java/com/mall/admin/entity/Product.java
package com.mall.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体类
 *
 * @author Mall Admin Team
 * @since 1.0.0
 */
@Data
@TableName("products")
@Schema(description = "商品实体")
public class Product {
    
    @TableId(value = "product_id", type = IdType.AUTO)
    @Schema(description = "商品ID")
    private Integer productId;
    
    @Schema(description = "商品名称")
    private String productName;
    
    @Schema(description = "商品编码")
    private String productCode;
    
    @Schema(description = "分类ID")
    private Integer categoryId;
    
    @Schema(description = "品牌")
    private String brand;
    
    @Schema(description = "商品描述")
    private String description;
    
    @Schema(description = "材质")
    private String material;
    
    @Schema(description = "颜色")
    private String color;
    
    @Schema(description = "尺寸")
    private String size;
    
    @Schema(description = "重量(kg)")
    private BigDecimal weight;
    
    @Schema(description = "原价")
    private BigDecimal originalPrice;
    
    @Schema(description = "现价")
    private BigDecimal currentPrice;
    
    @Schema(description = "成本价")
    private BigDecimal costPrice;
    
    @Schema(description = "促销价")
    private BigDecimal promotionPrice;
    
    @Schema(description = "库存数量")
    private Integer stockQuantity;
    
    @Schema(description = "销售数量")
    private Integer salesCount;
    
    @Schema(description = "浏览次数")
    private Integer viewCount;
    
    @Schema(description = "平均评分")
    private BigDecimal ratingAverage;
    
    @Schema(description = "评价数量")
    private Integer ratingCount;
    
    @Schema(description = "是否热门：0否，1是")
    private Integer isHot;
    
    @Schema(description = "是否新品：0否，1是")
    private Integer isNew;
    
    @Schema(description = "是否促销：0否，1是")
    private Integer isPromotion;
    
    @Schema(description = "促销开始时间")
    private LocalDateTime promotionStartTime;
    
    @Schema(description = "促销结束时间")
    private LocalDateTime promotionEndTime;
    
    @Schema(description = "商品状态：0下架，1上架")
    private Integer status;
    
    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updateTime;
}

File: vue3-admin/src/api/product.ts
/**
 * 商品相关API接口
 */
import request from '@/utils/request'

/**
 * 商品查询参数
 */
export interface ProductQueryParams {
  page?: number
  size?: number
  productName?: string
  productCode?: string
  categoryId?: number
  status?: number
  isHot?: number
  isNew?: number
  minPrice?: number
  maxPrice?: number
  brand?: string
  tags?: string
  keyword?: string
  createTimeStart?: string
  createTimeEnd?: string
  stockWarning?: number
}

/**
 * 商品图片信息
 */
export interface ProductImage {
  imageId: number
  productId: number
  imageUrl: string
  imageType: number // 1主图，2详情图
  sortOrder: number
  createTime: string
}

/**
 * 商品响应数据
 */
export interface ProductResponse {
  productId: number
  productName: string
  productCode: string
  categoryId: number
  categoryName?: string
  brand: string
  description: string
  material: string
  color: string
  size: string
  weight: number
  originalPrice: number
  currentPrice: number
  costPrice: number
  promotionPrice: number
  stockQuantity: number
  salesCount: number
  viewCount: number
  ratingAverage: number
  ratingCount: number
  // 图片相关字段
  imageUrl?: string // 主图URL
  images?: ProductImage[] // 所有图片列表
  mainImage?: string // 主图URL（兼容字段）
  isHot: number
  isNew: number
  isPromotion: number
  promotionStartTime?: string
  promotionEndTime?: string
  status: number
  createTime: string
  updateTime: string
}

/**
 * 分页响应数据
 */
export interface PageResponse<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

/**
 * 商品请求数据
 */
export interface ProductRequest {
  productName: string
  productCode?: string
  categoryId: number
  brand?: string
  description?: string
  material?: string
  color?: string
  size?: string
  weight?: number
  originalPrice: number
  currentPrice: number
  costPrice?: number
  promotionPrice?: number
  stockQuantity: number
  isHot?: number
  isNew?: number
  isPromotion?: number
  promotionStartTime?: string
  promotionEndTime?: string
  status: number
}

/**
 * 商品API类
 */
export class ProductAPI {
  /**
   * 获取商品列表（分页）
   */
  static async getProductList(params: ProductQueryParams): Promise<PageResponse<ProductResponse>> {
    const response = await request.get('/products', { params })
    console.log('获取商品列表（分页）', response)
    return response.data
  }

  /**
   * 根据ID获取商品详情
   */
  static async getProductById(id: number): Promise<ProductResponse> {
    const response = await request.get(`/products/${id}`)
    return response.data
  }

  /**
   * 创建商品
   */
  static async createProduct(data: ProductRequest): Promise<ProductResponse> {
    const response = await request.post('/products', data)
    return response.data
  }

  /**
   * 更新商品
   */
  static async updateProduct(id: number, data: ProductRequest): Promise<ProductResponse> {
    const response = await request.put(`/products/${id}`, data)
    return response.data
  }

  /**
   * 删除商品
   */
  static async deleteProduct(id: number): Promise<void> {
    await request.delete(`/products/${id}`)
  }

  /**
   * 批量删除商品
   */
  static async batchDeleteProducts(ids: number[]): Promise<void> {
    await request.delete('/products/batch', { data: { ids } })
  }

  /**
   * 更新商品状态
   */
  static async updateProductStatus(id: number, status: number): Promise<void> {
    await request.patch(`/products/${id}/status`, { status })
  }

  /**
   * 批量更新商品状态
   */
  static async batchUpdateProductStatus(ids: number[], status: number): Promise<void> {
    await request.patch('/products/batch/status', { ids, status })
  }

  /**
   * 获取热门商品
   */
  static async getHotProducts(limit?: number): Promise<ProductResponse[]> {
    const response = await request.get('/products/hot', { params: { limit } })
    return response.data
  }

  /**
   * 获取推荐商品
   */
  static async getRecommendedProducts(limit?: number): Promise<ProductResponse[]> {
    const response = await request.get('/products/recommended', { params: { limit } })
    return response.data
  }

  /**
   * 搜索商品
   */
  static async searchProducts(keyword: string, params?: ProductQueryParams): Promise<PageResponse<ProductResponse>> {
    const response = await request.get('/products/search', { 
      params: { keyword, ...params } 
    })
    return response.data
  }

  /**
   * 获取库存预警商品
   */
  static async getLowStockProducts(threshold?: number): Promise<ProductResponse[]> {
    const response = await request.get('/products/low-stock', { params: { threshold } })
    return response.data
  }
}

// 导出默认实例方法
export const productAPI = {
  getProductList: ProductAPI.getProductList.bind(ProductAPI),
  getProductById: ProductAPI.getProductById.bind(ProductAPI),
  createProduct: ProductAPI.createProduct.bind(ProductAPI),
  updateProduct: ProductAPI.updateProduct.bind(ProductAPI),
  deleteProduct: ProductAPI.deleteProduct.bind(ProductAPI),
  batchDeleteProducts: ProductAPI.batchDeleteProducts.bind(ProductAPI),
  updateProductStatus: ProductAPI.updateProductStatus.bind(ProductAPI),
  batchUpdateProductStatus: ProductAPI.batchUpdateProductStatus.bind(ProductAPI),
  getHotProducts: ProductAPI.getHotProducts.bind(ProductAPI),
  getRecommendedProducts: ProductAPI.getRecommendedProducts.bind(ProductAPI),
  searchProducts: ProductAPI.searchProducts.bind(ProductAPI),
  getLowStockProducts: ProductAPI.getLowStockProducts.bind(ProductAPI)
}

export default productAPI